src/pool/srv_pool.c

```diff
@@ -3688,7 +3698,12 @@ ds_pool_connect_handler(crt_rpc_t *rpc, int handler_version)
 	if (prop)
 		daos_prop_free(prop);
 out_svc:
-	ds_rsvc_set_hint(&svc->ps_rsvc, &out->pco_op.po_hint);
+	/*
+	 * NOTE: For ds_pool_skip_for_check() true case, after PS restart, current rank may be not
+	 *	 PS leader, do not set the hint under such case (rc == -DER_BUSY && nhandles == 0).
+	 */
+	if (rc != -DER_BUSY || nhandles != 0)
+		ds_rsvc_set_hint(&svc->ps_rsvc, &out->pco_op.po_hint);
```

Comment:
Because the new added ds_pool_skip_for_check(), we may be out from there. Under such case, it is unnecessary to set_hint.

Reply:
@Nasf-Fan, I'm interested in knowing why you think setting the hint is unnecessary. Hence, the question above asks what would happen if we did set the hint.

Reply:
If we call set_hint under my mentioned case, we may return some wrong hint to the sponsor? Not sure, depends on the detailed RDB logic. So I'm trying to keep the logic as much the same as original case.

Reply:
Since an `rsvc_hint` is not a ping, returning a hint in the case you mentioned would not cause any delay to the `rsvc_client`. Also, the hint would not be wrong: It says the leader of term x is Lucy, which remains true even when Lily becomes the leader of term y, y > x. So I think the PR should not make this change.

Reply:
Since you confirm that, we can consider to cleanup related logic in subsequent patch.